src/foundry/common/documents/tile.mjs.d.ts

```diff
@@ -0,0 +1,183 @@
+  /**
+   * @param data    - Initial data from which to construct the Tile
+   * @param context - Construction context options
+   */
+  constructor(data: BaseTile.ConstructorData, context?: DocumentConstructionContext);
+}
+export default BaseTile;
+
+declare namespace BaseTile {
+  type Metadata = Merge<
+    DocumentMetadata,
+    { name: "Tile"; collection: "tiles"; label: "DOCUMENT.Tile"; labelPlural: "DOCUMENT.Tiles" }
+  >;
+
+  type SchemaField = fields.SchemaField<Schema>;
+  type ConstructorData = UpdateData;
+  type UpdateData = fields.SchemaField.InnerAssignmentType<Schema>;
+  type Properties = fields.SchemaField.InnerInitializedType<Schema>;
+  type Source = fields.SchemaField.InnerPersistedType<Schema>;
+
+  interface Schema extends DataSchema {
+    /**
+     * The _id which uniquely identifies this Tile embedded document
+     * @defaultValue `null`
+     */
+    _id: fields.DocumentIdField;
+
+    /**
+     * An image or video texture which this tile displays.
+     * @defaultValue `null`
+     */
+    texture: TextureData<{}, { categories: ["IMAGE", "VIDEO"]; initial: null; wildcard: false }>;
+
+    /**
+     * The pixel width of the tile
+     * @defaultValue `0`
+     */
+    width: fields.NumberField<{
+      required: true;
+      min: 0;
+      nullable: false;
+      step: 0.1;
+    }>;
+
+    /**
+     * The pixel height of the tile
+     * @defaultValue `0`
+     */
+    height: fields.NumberField<{ required: true; min: 0; nullable: false; step: 0.1 }>;
+
+    /**
+     * The x-coordinate position of the top-left corner of the tile
+     * @defaultValue `0`
+     */
+    x: fields.NumberField<{ required: true; integer: true; nullable: false; initial: 0; label: "XCoord" }>;
+
+    /**
+     * The y-coordinate position of the top-left corner of the tile
+     * @defaultValue `0`
+     */
+    y: fields.NumberField<{ required: true; integer: true; nullable: false; initial: 0; label: "YCoord" }>;
+
+    /**
+     * The z-index ordering of this tile relative to its siblings
+     * @defaultValue `100`
+     */
+    z: fields.NumberField<{ required: true; integer: true; nullable: false; initial: 100 }>;
+
+    /**
+     * The angle of rotation for the tile between 0 and 360
+     * @defaultValue `0`
+     */
+    rotation: fields.AngleField;
+
+    /**
+     * The tile opacity
+     * @defaultValue `1`
+     */
+    alpha: fields.AlphaField;
+
+    /**
+     * Is the tile currently hidden?
+     * @defaultValue `false`
+     */
+    hidden: fields.BooleanField;
+
+    /**
+     * Is the tile currently locked?
+     * @defaultValue `false`
+     */
+    locked: fields.BooleanField;
+
+    /**
+     * Is the tile an overhead tile?
+     * @defaultValue `false`
+     */
+    overhead: fields.BooleanField;
+
+    /**
+     * Is the tile a roof?
+     * @defaultValue `false`
+     */
+    roof: fields.BooleanField;
+
+    /**
+     * The tile's occlusion settings
+     * @defaultValue `null`
```

Comment:
This is not quite correct. Most fields have default initial values. You can use the hover feature on the different fields. All of the fields have their default initial values documented.

Reply:
In general, are `fields.SchemaField` entries that exist on other classes going to be non-null by default?

I hovered over this code in the types project and in the original Foundry script for `tile.mjs`, but I had some trouble determining the default value.

I eventually went into the runtime and checked the default value after dropping in a tile.

What is the main clue to look for there?

Reply:
Aha! I see now: 

![image](https://github.com/League-of-Foundry-Developers/foundry-vtt-types/assets/4632129/cdcfbf78-74e8-4560-98d1-e703a81c85cf)

I am getting there slowly 😅

Reply:
Yeah as mentioned in the discord, it seems I need to reformat those doc comments,so the rendered markdown output renders better.